rust/agents/relayer/src/msg/gas_payment/mod.rs

```diff
@@ -80,15 +87,18 @@ impl GasPaymentEnforcer {
         &self,
         message: &HyperlaneMessage,
         tx_cost_estimate: &TxCostEstimate,
-    ) -> Result<Option<U256>> {
+    ) -> Result<GasPolicyStatus> {
         let msg_id = message.id();
         let gas_payment_key = GasPaymentKey {
             message_id: msg_id,
             destination: message.destination,
         };
-        let current_payment = self
+        let Some(current_payment) = self
             .db
-            .retrieve_gas_payment_by_gas_payment_key(gas_payment_key)?;
+            .retrieve_gas_payment_by_gas_payment_key(gas_payment_key)?
+        else {
+            return Ok(GasPolicyStatus::NoPaymentFound);
```

Comment:
does this break the use case of configuring the relayer with GasPaymentEnforcementPolicy::None?

I think we'll want to keep the ability to configure that for some / all messages

Reply:
hmm good point. Changed it to return:
- `PolicyMet` if the policy passes, regardless of there being a gas payment or not
- `NoPaymentFound`, if the policy rejects it and there is no gas payment
- `PolicyNotMet`, if the policy rejects it in spite of there being a gas payment